internal/dependencymanager/dependencyinstaller.go

```diff
@@ -0,0 +1,249 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dependencymanager
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/onflow/flow-cli/internal/util"
+
+	"github.com/onflow/flow-cli/flowkit/gateway"
+
+	"github.com/onflow/flow-cli/flowkit/project"
+
+	flowsdk "github.com/onflow/flow-go-sdk"
+
+	"github.com/onflow/flow-cli/flowkit/config"
+
+	"github.com/onflow/flow-cli/flowkit"
+	"github.com/onflow/flow-cli/flowkit/output"
+)
+
+type DependencyInstaller struct {
+	Gateways map[string]gateway.Gateway
+	Logger   output.Logger
+	State    *flowkit.State
+	Mutex    sync.Mutex
+}
+
+func NewDepdencyInstaller(logger output.Logger, state *flowkit.State) *DependencyInstaller {
+	emulatorGateway, err := gateway.NewGrpcGateway(config.EmulatorNetwork)
+	if err != nil {
+		logger.Error(fmt.Sprintf("Error creating emulator gateway: %v", err))
```

Comment:
we are logging the error but still contiuning down the chain of the funciton, is that intended?